sale_properties_dynamic_fields/sale_order.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014-15 Agile Business Group sagl
+#    (<http://www.agilebg.com>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm
+from lxml import etree
+from openerp.tools.translate import _
+
+
+class SaleOrder(orm.Model):
+    _inherit = 'sale.order'
+
+    def fields_view_get(
+        self, cr, uid, view_id=None, view_type='form', context=None,
+        toolbar=False, submenu=False
+    ):
+        res = super(SaleOrder, self).fields_view_get(
+            cr, uid, view_id=view_id, view_type=view_type, context=context,
+            toolbar=toolbar, submenu=submenu)
+        if res['name'] == u'sale.order.form':
+            property_group_pool = self.pool['mrp.property.group']
+            group_to_draw_ids = property_group_pool.search(cr, uid, [
+                ('draw_dynamically', '=', True),
+            ], context=context)
+            if group_to_draw_ids:
```

Comment:
consider extracting the body of the test to a separate method with a well chosen name to improve readability.